generator.yaml

```diff
@@ -132,6 +132,49 @@ resources:
         from:
           operation: DescribeTransformJob
           path: FailureReason
+  DataQualityJobDefinition:
+    exceptions:
+      errors:
+          404:
+            code: ResourceNotFound
+            message_prefix: Monitoring Job Definition
+  ModelQualityJobDefinition:
+    exceptions:
+      errors:
+          404:
+            code: ResourceNotFound
+            message_prefix: Monitoring Job Definition
+  ModelBiasJobDefinition:
+    exceptions:
+      errors:
+          404:
+            code: ResourceNotFound
+            message_prefix: Monitoring Job Definition
+  ModelExplainabilityJobDefinition:
+    exceptions:
+      errors:
+          404:
+            code: ResourceNotFound
+            message_prefix: Monitoring Job Definition
+  MonitoringSchedule:
+    fields:
+      MonitoringScheduleStatus:
+        is_read_only: true
+        is_printable: true
+        from:
+          operation: DescribeMonitoringSchedule
+          path: MonitoringScheduleStatus
+      FailureReason:
```

Comment:
Why would a schedule be in a Pending or Failed Status and how would customer recover from failures

We need to treat this as a job and not like config

Reply:
I disagree. The only commonality between this and a job is that it requires constant reconciliation to update the status fields. Otherwise it behaves exactly like a configuration. `Pending` or `Failed` statuses indicate that they have an error in their job definition, which caused the processing job running the model monitoring task to fail. For a user this would require updating their job definition - it has no effect on the scheduling. These status fields are conveniences to understand the previous state of your model monitoring invocations, but not the status of your schedule itself.

Reply:
> Pending or Failed status indicate that they have an error in their job definition, which caused the processing job running the model monitoring task to fail

I think this is not correct. the schedule itself can fail to register or update independent of the processing job.
FailureReason of the monitoring schedule is independent of the processing job.

If I understand if correctly, create_monitoring_schedule will return success and put the schedule to `Pending` status and user need to do a `describe_` until it is in `Scheduled`

and user also has the ability to update the monitoring schedule in which case there will be a similar lifecycle